packages/common/amplify_db_common/lib/src/connect_io.dart

```diff
@@ -20,10 +22,11 @@ import 'package:path_provider/path_provider.dart';
 /// {@macro amplify_db_common.connect}
 QueryExecutor connect({
   required String name,
-  String? path,
+  FutureOr<String>? path,
 }) {
   return LazyDatabase(() async {
-    if (path == null) {
+    final resolvedPath = await path;
```

Comment:
It seems the future itself is null so can we check if it's not null first before awaiting on it?

Reply:
`await null` is legit.

Reply:
alright thanks guys!